src/main/java/query/GetTableQuery.java

```diff
@@ -0,0 +1,44 @@
+package query;
+
+import static org.example.models.Tables.*;
+
+import org.example.models.tables.records.CourseRecord;
+import org.example.models.tables.records.DegreeProgramRecord;
+import org.jooq.DSLContext;
+
+public class GetTableQuery {
+  private GetTableQuery() {}
+  ;
+
+  public static String degreeProgrammTable(DSLContext sql) {
```

Comment:
... and replace / rename this to `execute` (implementing the `Query` interface) and make it return a `List<String>` containing the courses. You're mixing application logic with presentation logic here. Building the string (= the "presentation" of the data) in the `Screen` would make more sense imo.